### PR TITLE
Added support for Key vault service version `7.5-preview.1` and prepared release.

### LINKED ISSUE #36776: Regenerate KeyVault SDK using the latest swagger version
The fix for key rotation for code generators has been merged. Use commit 5dd1107d5f2be8d600325d795450e1d854fbe7e8 for codegen

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes #36776.

Also disabled the key release tests while we wait for a service fix.

### CODE REVIEW

**Review by mccoyp — COMMENTED** [review]
Is it possible to add a mocked test that just verifies the intended behavior of `useManagedIdentity` (in how things get constructed, rather than with a service interaction)?

File: sdk/keyvault/azure-security-keyvault-administration/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
 
 ### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+- Added support for service version `7.5-preview.1`.
+- Managed Identity can now be used in place of a SAS token to access the blob storage resource when performing backup and restore operations using a `KeyVaultBackupClient` or `KeyVaultBackupAsyncClient`. This is now the default behavior if a `null` SAS token is provided to the `beginBackup()`, `beginRestore()` or `beginSelectiveRestore()` methods.
```
Comment (pallavit) [inline sdk/keyvault/azure-security-keyvault-administration/CHANGELOG.md:7]: Is this a bugfix?

Comment (pallavit) [reply sdk/keyvault/azure-security-keyvault-administration/CHANGELOG.md:7]: Oh yeah this is the admin one!

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java [hunk 2]
```diff
@@ -81,9 +80,7 @@ public RoleAssignmentsImpl getRoleAssignments() {
      */
     public KeyVaultAccessControlClientImpl(String apiVersion) {
         this(
-                new HttpPipelineBuilder()
-                        .policies(new UserAgentPolicy(), new RetryPolicy(), new CookiePolicy())
-                        .build(),
+                new HttpPipelineBuilder().policies(new UserAgentPolicy(), new RetryPolicy()).build(),
```
Comment (pallavit) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java:83]: Why is cookie policy removed?

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java:83]: I don't know, this change was made by AutoRest. Is this a recent change for our client impls @alzimmermsft, @srnagar?

Comment (heaths) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java:83]: FWIW, I don't know why we'd even want to support cookies for REST API access. The service shouldn't be setting any anyway.

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java [hunk 3]
```diff
@@ -87,9 +87,6 @@ public RoleAssignment setProperties(RoleAssignmentPropertiesWithScope properties
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
-        jsonWriter.writeStringField("id", this.id);
-        jsonWriter.writeStringField("name", this.name);
-        jsonWriter.writeStringField("type", this.type);
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java:92]: This is sus.

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleDefinition.java [hunk 4]
```diff
@@ -192,9 +192,6 @@ public RoleDefinition setAssignableScopes(List<KeyVaultRoleScope> assignableScop
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
-        jsonWriter.writeStringField("id", this.id);
-        jsonWriter.writeStringField("name", this.name);
-        jsonWriter.writeStringField("type", Objects.toString(this.type, null));
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleDefinition.java:197]: This is sus.

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/SettingsListResult.java [hunk 5]
```diff
@@ -35,7 +35,6 @@ public List<Setting> getSettings() {
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
-        jsonWriter.writeArrayField("settings", this.settings, (writer, element) -> writer.writeJson(element));
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/SettingsListResult.java:38]: This is sus.

File: sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md [hunk 6]
```diff
@@ -64,7 +64,7 @@ include-read-only-in-constructor-args: true
 These settings apply only when `--tag=rbac` is specified on the command line.
 
 ``` yaml $(tag) == 'rbac'
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/551275acb80e1f8b39036b79dfc35a8f63b601a7/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/rbac.json
+input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/6a2e3c7617314fe4ea7e5706da5437214e8a602b/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.5-preview.1/rbac.json
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md:67]: You need to use a2f6f742d088dcc712e67cb2745d8271eaa370ff

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java [hunk 7]
```diff
@@ -30,7 +30,7 @@ public BackupSecretResult() {}
      */
     public byte[] getValue() {
         if (this.value == null) {
-            return new byte[0];
+            return null;
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java:33]: This is a break - albeit it better - change in behavior. Is this okay?

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultAccessControlClientImpl.java:83]: Yeah. Removed Cookie after discussed with Srikanta.

I don't know any SDK that really need it. But do let me know if I am wrong.

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java (hunk 7 above)
Comment (weidongxu-microsoft) [reply sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java:33]: setting `null-byte-array-maps-to-empty-array: true` in swagger/readme.md should revert this change (if we don't want this behavior break)

@alzimmermsft

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java:92]: This was a change in code generation logic to stop serializing response only values, Effectively, this makes this the same as how Jackson Databind would have handled serialization.

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java (hunk 7 above)
Comment (alzimmermsft) [reply sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java:33]: This was done as `new byte[0]` causes issues sometimes in logic that expected null.

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java (hunk 3 above)
Comment (heaths) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/RoleAssignment.java:92]: Verified it's not needed to create role assignments (or defs below). Thanks for explaining!

File: sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java (hunk 7 above)
Comment (heaths) [reply sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/models/BackupSecretResult.java:33]: It makes more sense, but should it go into the CHANGELOG.md? Not blocking, and given your explanation of the other concerns, I'll sign off. Do with this what you will.

**Review by heaths — APPROVED** [review]

**Pull request merged**